main/src/main/scala/sbt/internal/nio/CheckBuildSources.scala

```diff
@@ -8,56 +8,153 @@
 package sbt
 package internal.nio
 
-import sbt.Keys.{ baseDirectory, state, streams }
+import java.nio.file.Path
+import java.util.concurrent.atomic.{ AtomicBoolean, AtomicReference }
+import sbt.BasicCommandStrings.{ RebootCommand, TerminateAction }
+import sbt.Keys.{ baseDirectory, pollInterval, state }
+import sbt.Scope.Global
 import sbt.SlashSyntax0._
+import sbt.internal.CommandStrings.LoadProject
+import sbt.internal.SysProp
+import sbt.internal.util.AttributeKey
 import sbt.io.syntax._
 import sbt.nio.FileChanges
+import sbt.nio.FileStamp
 import sbt.nio.Keys._
-import sbt.nio.file.{ Glob, ** }
+import sbt.nio.file.{ FileAttributes, FileTreeView, Glob, ** }
 import sbt.nio.file.syntax._
+import sbt.nio.Settings
 
 import scala.annotation.tailrec
+import scala.concurrent.duration.{ Deadline => SDeadline, _ }
+
+/**
+ * This class is used to determine whether sbt needs to automatically reload
+ * the build because its source files have changed. In general, it will use
+ * a FileTreeRepository to monitor the build source directories and it will
+ * only actually check whether any sources have changed if the monitor has
+ * detected any events. Because it's using asynchronous monitoring by default,
+ * the automatic reloading should not be relied upon in batch scripting. It is
```

Comment:
In general though does this feature matter unless the user is in an interactive session, like `shell` or `~`?
Is it possible to not have this turned on at all during scripted test or CI processing etc?

Reply:
It could certainly be disabled. The overhead is now low enough that I don't think it really matters.

Reply:
Even before, the time spent in CheckBuildSources was surely a very small component of the total runtime. But it was noticeable that when you did a no-op enter in the console that you wouldn't get re-prompted immediately.